invenio_sip2/api_views.py

```diff
@@ -0,0 +1,129 @@
+# the Free Software Foundation, version 3 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Blueprint for Invenio-SIP2."""
+
+from __future__ import absolute_import, print_function
+
+from flask import Blueprint, jsonify
+
+from .records.record import Client, Server
+
+api_blueprint = Blueprint(
+    'api_sip2',
+    __name__,
+    url_prefix='/monitoring/sip2'
+)
+
+
+@api_blueprint.route('/status')
+def status():
+    """Displa status for all SIP2 server."""
+    return jsonify(Monitoring.status())
+
+
+@api_blueprint.route('/servers')
+def get_servers():
+    """Display all running SIP2 servers."""
+    try:
+        servers = Server.get_all_records()
+        print('servers:', servers)
+        return jsonify({'servers': servers})
+    except Exception as error:
+        raise error
+        return jsonify({'ERROR': str(error)})
+
+
+@api_blueprint.route('/servers/<string:server_id>')
+def get_server(server_id):
+    """Display all running SIP2 servers."""
+    try:
+        server = Server.get_record_by_id(server_id)
+        server['clients'] = Monitoring.get_list_of_clients_by_server(server_id)
+        return jsonify({
+            'id': server.id,
+            'metadata': server,
+        })
+    except Exception as error:
+        raise error
+        return jsonify({'ERROR': str(error)})
+
+
+@api_blueprint.route('/clients')
+def get_clients():
+    """Display all connected clients to server."""
+    try:
+        clients = Client.get_all_records()
+    except Exception as error:
+        raise error
+        return jsonify({'ERROR': str(error)})
+    return jsonify({'clients': clients})
+
+
+class Monitoring:
+    """Monitoring class."""
+
+    @classmethod
+    def status(cls):
+        """Check status for all servers."""
+        result = {}
+        servers = cls.get_list_of_servers()
+        result['servers'] = len(servers)
+        result['clients'] = cls.get_number_of_clients()
+        result['status'] = 'green'
+        if result['servers'] > 0:
+            info = {}
+            for server in cls.get_list_of_servers():
+                info[server.id] = {}
+                info[server.id]['status'] = server.get('status')
+                info[server.id]['nb_client'] = \
+                    cls.get_number_client_by_server(server.id)
+                if info[server.id]['status'] == 'down':
+                    result['status'] = 'red'
+                result['servers_info'] = info
+        return result
+
+    @classmethod
+    def get_count_servers(cls):
+        """Get total number of servers."""
+        servers = Server.get_all_records()
```

Comment:
Note :: Don't know if this call consume a huge performance or not ? If yes, maybe just check into the datastore rather than load servers object just to count the list length ? Into `Server.create` you call is server is already running, ... is it necessary here ? 

Maybe just return something like `len(datastore.all(Server.rercord_type))` or a specific count method into Sip2Metadata class

Reply:
Ok. It's done